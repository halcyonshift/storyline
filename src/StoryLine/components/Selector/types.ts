export type SelectorProps = {
    onClick: (value: string) => void
}
