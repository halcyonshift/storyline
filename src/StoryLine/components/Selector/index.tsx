import { ReactElement, useState } from 'react'
import { Box, IconButton, Tooltip } from '@mui/material'
import { useTranslation } from 'react-i18next'
import { CHARACTER_ICONS, ITEM_ICONS, LOCATION_ICONS, NOTE_ICONS } from '@sl/constants/icons'
import { ColorType } from '@sl/theme/types'
import { getHex } from '@sl/theme/utils'
import { SelectorProps } from './types'

const Selector = ({ onClick }: SelectorProps) => {
    const [active, setActive] = useState<string>('character')
    const { t } = useTranslation()

    const handleClick = (table: string) => {
        setActive(table)
        onClick(table)
    }

    const buttons: { table: string; color: ColorType; icon: ReactElement }[] = [
        { table: 'character', color: 'emerald', icon: CHARACTER_ICONS.character },
        { table: 'location', color: 'amber', icon: LOCATION_ICONS.location },
        { table: 'item', color: 'purple', icon: ITEM_ICONS.item },
        { table: 'note', color: 'sky', icon: NOTE_ICONS.note }
    ]

    return (
        <Box className='flex justify-around'>
            {buttons.map((button) => (
                <Box className={button.table === active ? 'opacity-100' : 'opacity-50'}>
                    <Tooltip title={t(`component.separator.${button.table}`)}>
                        <IconButton
                            key={button.table}
                            onClick={() => handleClick(button.table)}
                            sx={{ color: getHex(button.color, 600) }}>
                            {button.icon}
                        </IconButton>
                    </Tooltip>
                </Box>
            ))}
        </Box>
    )
}

export default Selector
