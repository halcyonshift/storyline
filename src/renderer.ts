import { init } from '@sentry/electron/renderer'
import { init as reactInit } from '@sentry/react'

if (parseInt(process.env.DEBUG) === 0) {
    init({ dsn: process.env.SENTRY }, reactInit)
}

import '@fontsource/roboto/300.css'
import '@fontsource/roboto/400.css'
import '@fontsource/roboto/500.css'
import '@fontsource/roboto/700.css'
import './index.css'
import './StoryLine'
