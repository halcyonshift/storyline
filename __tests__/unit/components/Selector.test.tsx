import userEvent from '@testing-library/user-event'
import renderer from 'react-test-renderer'
import Selector from '@sl/components/Selector'
import { render, screen } from '../test-utils'

const onClick = jest.fn()

describe('<Selector />', () => {
    beforeAll(() => {
        render(<Selector onClick={onClick} />)
    })

    it('it updates active table on click', async () => {
        const buttons = screen.getAllByRole('button')

        await userEvent.click(buttons[0])
        expect(onClick).toHaveBeenCalledWith('character')

        await userEvent.click(buttons[1])
        expect(onClick).toHaveBeenCalledWith('location')

        await userEvent.click(buttons[2])
        expect(onClick).toHaveBeenCalledWith('item')

        await userEvent.click(buttons[3])
        expect(onClick).toHaveBeenCalledWith('note')
    })

    it('renders correctly', () => {
        const tree = renderer.create(<Selector onClick={onClick} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
