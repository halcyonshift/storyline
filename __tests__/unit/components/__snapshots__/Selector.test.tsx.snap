// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Selector /> renders correctly 1`] = `
<div
  className="flex justify-around MuiBox-root css-0"
>
  <div
    className="opacity-100 MuiBox-root css-0"
  >
    <button
      aria-label="component.separator.character"
      aria-labelledby={null}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-17lsm5k-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element={true}
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        aria-label="Group"
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="GroupOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M9 13.75c-2.34 0-7 1.17-7 3.5V19h14v-1.75c0-2.33-4.66-3.5-7-3.5zM4.34 17c.84-.58 2.87-1.25 4.66-1.25s3.82.67 4.66 1.25H4.34zM9 12c1.93 0 3.5-1.57 3.5-3.5S10.93 5 9 5 5.5 6.57 5.5 8.5 7.07 12 9 12zm0-5c.83 0 1.5.67 1.5 1.5S9.83 10 9 10s-1.5-.67-1.5-1.5S8.17 7 9 7zm7.04 6.81c1.16.84 1.96 1.96 1.96 3.44V19h4v-1.75c0-2.02-3.5-3.17-5.96-3.44zM15 12c1.93 0 3.5-1.57 3.5-3.5S16.93 5 15 5c-.54 0-1.04.13-1.5.35.63.89 1 1.98 1 3.15s-.37 2.26-1 3.15c.46.22.96.35 1.5.35z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="opacity-50 MuiBox-root css-0"
  >
    <button
      aria-label="component.separator.location"
      aria-labelledby={null}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1kj6x23-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element={true}
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        aria-label="LocationOn"
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="LocationOnOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zM7 9c0-2.76 2.24-5 5-5s5 2.24 5 5c0 2.88-2.88 7.19-5 9.88C9.92 16.21 7 11.85 7 9z"
        />
        <circle
          cx="12"
          cy="9"
          r="2.5"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="opacity-50 MuiBox-root css-0"
  >
    <button
      aria-label="component.separator.item"
      aria-labelledby={null}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ldsnqk-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element={true}
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        aria-label="Category"
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="CategoryOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m12 2-5.5 9h11L12 2zm0 3.84L13.93 9h-3.87L12 5.84zM17.5 13c-2.49 0-4.5 2.01-4.5 4.5s2.01 4.5 4.5 4.5 4.5-2.01 4.5-4.5-2.01-4.5-4.5-4.5zm0 7c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zM3 21.5h8v-8H3v8zm2-6h4v4H5v-4z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="opacity-50 MuiBox-root css-0"
  >
    <button
      aria-label="component.separator.note"
      aria-labelledby={null}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1a4ila1-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element={true}
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        aria-label="StickyNote2"
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="StickyNote2OutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v9h-5v5H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h10l6-6V5c0-1.1-.9-2-2-2zm-7 11H7v-2h5v2zm5-4H7V8h10v2z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
